1. Print out a data type of a variable?

In [1]:
print (type ('Hello friend'))
print (type (123.9897))

<class 'str'>
<class 'float'>


2. Round 4.5667 to the nearest hundreth using a build-in function, round().

In [4]:
digit = 4.5667
print (round(digit,2))

4.57


3. Convert "657" to an integer data type

In [5]:
print (int('57'))

57


4. Write a function to calculate the miles per gallon.
Get a miles driven and gallons used from a user and call your function to calculate the miles per gallon. 

In [16]:
miles = float (input("Enter Miles driven: "))
gaseUsed = float (input("Enter Gallon used:  "))

mpg =  float (miles / gaseUsed )
print ( float(mpg))


Enter Miles driven: 150
Enter Gallon used:  15
10.0


5. Enhance your function from exercise 3. Additionally, get a gas price per gallon from a user and write a program to compute the total gas cost and cost per mile. 

The output should look something like this:

    Enter miles driven:            150
    Enter gallons of gas used:     15
    Enter cost per gallon:          3

    Miles Per Gallon:   10.0
    Total Gas Cost:     45.0
    Cost per Mile:      0.3


In [18]:
miles = float (input("Enter Miles driven: "))
gasUsed = float (input ("Enter Gallon used:  "))
costPerGalon  = float (input("Enter cost per gallon: "))
costPerMile = float (costPerGalon / (miles / gaseUsed ))

mpg =  float (miles / gaseUsed )
print (f"\nMiles Per Gallon:  ", mpg)
print (f"Total Gas Cost:    ", costPerGalon*gasUsed)
print (f"Cost per Mile:  ", costPerMile )


Enter Miles driven: 150
Enter Gallon used:  15
Enter cost per gallon: 3

Miles Per Gallon:   10.0
Total Gas Cost:     45.0
Cost per Mile:   0.3


6. Write a function called multiply_two() that accepts two parameters. The function should print a message like '15 + 2 = 17'.  Randomly generate two numbers between 1 and 100, then call the function with the two random numbers.

In [28]:
def  multiply_two( x=1, y=1) :    
    area = x +  y    
    return area

num1 = int (input("Enter 1st Number: "))
num2 = int (input("Enter 2nd Number: "))

print ( num1, " + ", num2 , " = " , multiply_two (num1, num2) )

Enter 1st Number: 15
Enter 2nd Number: 2
15  +  2  =  17


7. Write a function called make_shirt() that accepts a size and the text of a message. The function should print a sentence summarizing the size of the shirt and the message printed on it. The default shirt size is large and default message is 'I love Python!' Make a default size shirt with a message, a medium shirt with the default message, and a shirt of any size with a different message.

In [39]:
def  make_shirt( size="Large", msg="I love Python!") :    
    s = input("Enter shirt size you want: ")
    m = input("Enter message to the shirt you like: ")
    print (size, " size ",msg, "\n")
    return

make_shirt() 
make_shirt ("Medium","")
make_shirt ("Small", "I love sunshine")

Enter shirt size you want: 
Enter message o the shirt you like: 
Large  size  I love Python! 

Enter shirt size you want: Medium
Enter message o the shirt you like: 
Medium  size   

Enter shirt size you want: Small
Enter message o the shirt you like: I lovesunsh
Small  size  I love sunshine 



8. Create a function for a multiplication table.  Get a number from a user and then call you function which produces a multiplication table for that number.
The output should look something like this:

        Enter a number: 3
        3 X 1 = 3
        3 X 2 = 6
        3 X 3 = 9
        3 X 4 = 12
        3 X 5 = 15
        3 X 6 = 18
        3 X 7 = 21
        3 X 8 = 24
        3 X 9 = 27
        

In [60]:
def  multiplication_table( num, times) : 
    num = int(num)*int(times)
    return num


num = input("Enter a number: ")
print (num, " x ", 1, " = ", multiplication_table(num, 1))
print (num, " x ", 2, " = ", multiplication_table(num, 2))
print (num, " x ", 3, " = ", multiplication_table(num, 3))
print (num, " x ", 4, " = ", multiplication_table(num, 4))
print (num, " x ", 5, " = ", multiplication_table(num, 5))
print (num, " x ", 6, " = ", multiplication_table(num, 6))
print (num, " x ", 7, " = ", multiplication_table(num, 7))
print (num, " x ", 8, " = ", multiplication_table(num, 8))
print (num, " x ", 9, " = ", multiplication_table(num, 9))
print (num, " x ", 10, " = ", multiplication_table(num, 10))
print (num, " x ", 11, " = ", multiplication_table(num, 11))
print (num, " x ", 12, " = ", multiplication_table(num, 12))
 

Enter a number: 3
3  x  1  =  3
3  x  2  =  6
3  x  3  =  9
3  x  4  =  12
3  x  5  =  15
3  x  6  =  18
3  x  7  =  21
3  x  8  =  24
3  x  9  =  27
3  x  10  =  30
3  x  11  =  33
3  x  12  =  36


9. Write a function that accepts a string and returns a string that is 
    - without any leading nor trailing blanks
    - converts to all capital letters
    - and replaces all blanks with a period(.).  

Write a program to accept a string from a user then call your function and display the original string and changed string.  

The output should look something like this:

        Enter a text: may your dreams come true
        
        Original Text: may your dreams come true
        New Text: MAY.YOUR.DREAMS.COME.TRUE

In [74]:
def  multiplication_table( inStr) :
     outStr = inStr.strip()
     outStr = outStr.upper()
     outStr = outStr.replace(" ", '.')
     return outStr


word = input("Enter a string: ")
print (multiplication_table( word))

Enter a string: may your dreams come true
MAY.YOUR.DREAMS.COME.TRUE


10. Write a function that returns a randomly selected character from a passed string. Write a program to accept a string from a user then call your function and display the original string and selected character.

The output should look something like this:

        Enter a text: functions
        Original Text:	functions
        Random character:	s

In [93]:
import random as rand
def get_rand_char(inStr1) :
    start = int(0)
    end = int(len(inStr1))
    r = rand.randint( start, end )
    outChr = inStr1[r]
    return outChr

word = input("Enter a string: ")
print (get_rand_char(word))

Enter a string: smooth
t
